src/entropies/renyi.jl

```diff
@@ -89,3 +94,14 @@ function entropy_renyi!(p, x, est; q = 1.0, α = nothing, base = MathConstants.e
     probabilities!(p, x, est)
     entropy_renyi(p; q = q, base = base)
 end
+
+"""
+    maxentropy_renyi(N::Int, base = MathConstants.e)
+
+Convenience function that computes the maximum value of the order-`q` generalized
+Rényi entropy for an `N`-element probability distribution, i.e.
+``\\log_{base}(N)``, which is useful for normalization when `N` is known.
+
+See also [`entropy_renyi`](@ref), [`entropy_normalized`](@ref).
+"""
+maxentropy_renyi(N::Int; base = MathConstants.e) = log(base, N)
```

Comment:
Why would we need such a function? I'm not convinced that providing a wrapper function over `log` is useful either :D

Reply:
Because maximum entropy in itself is a useful thing to be able to compute. For Renyi entropy, it is trivially `log(N)`, but for other entropies, it is not trivial at all. For stretched exponential entropy, for example, which I've prepared in a non-committed PR, the maximum entropy is a function of the parameters of the method and involves complicated expression involving gamma and incomplete gamma functions. 

I think there should be a convenient way for the user to compute this quantity, to also be able to work with the entropies from a theoretical standpoint. All analytical tests for this stretched exponential entropy, and the example reproducing the original paper, for example, involve using `maxentropy_stretched_exponential`. 

The maximum entropy is a fundamental property of entropies. I don't see why we shouldn't provide this functionality. This is after all a package that is dedicated specifically to computing entropies and related quantities.  I don't think pruning is necessary just because we don't see an immediate use for the method at the moment.

Reply:
If you also vote for that I'd like to do it after I merge this PR.

Reply:
> Actually, this whole business makes me think if we should make "entropy types". We can have a single entropy(...) function, and we have "entropy types" such as Renyi() and Tsallis(). So the renyi entropy for a given estimator would be entropy(Renyi(), x, est) This way we can actually call maximum(Renyi(), PermutationX()) instead of passing functions. The estimators can have arguments such as Renyi(q) for the order q.

Yes, going for the dispatch version would perhaps be better once the codebase grows. I wasn't aware to begin with how many types of entropies there actually are, but they are in fact plentiful.

If the rest of this PR looks ok to you, maybe we just merge this PR (in the current state) and the original one to `main`.  Then, as you say, we can implement the multiple dispatch `entropy(EntropyType(), Estimator())` version in a separate PR. 

If this is merged quickly, I can put together a PR tonight for the new version, because I'd like to get this off the table as soon as possible and arrive at a stable API.